packages/compass-aggregations/src/modules/pipeline.ts

```diff
@@ -441,20 +444,39 @@ export default function reducer(state = [emptyStage()], action: AnyAction): Stat
 /**
  * Action creator for adding a stage.
  */
-export const stageAdded = (): AnyAction => ({
-  type: STAGE_ADDED
-});
+export const stageAdded =
+  (): ThunkAction<void, RootState, void, AnyAction> => (dispatch, getState) => {
+    const { pipeline } = getState();
+    track('Aggregation Edited', {
+      num_stages: pipeline.length,
+      stage_action: 'stage_added',
+      stage_name: null
+    });
+    dispatch({
+      type: STAGE_ADDED
+    });
+  };
 
 /**
  * Action creator for adding a stage after current one.
  * @param {Number} index - The index of the stage.
  *
  * @returns {Object} the stage added after action.
  */
-export const stageAddedAfter = (index: number): AnyAction => ({
-  index: index,
-  type: STAGE_ADDED_AFTER
-});
+export const stageAddedAfter =
+  (index: number): ThunkAction<void, RootState, void, AnyAction> =>
+  (dispatch, getState) => {
+    const { pipeline } = getState();
+    track('Aggregation Edited', {
+      num_stages: pipeline.length,
+      stage_action: 'stage_added',
```

Comment:
Do we want this to be a separate event from the regular stage added?
```suggestion
      stage_action: 'stage_added_after',
```
I'm thinking it's probably more useful having them together as they currently are, I figured I'd clarify though as they are separate ui items/actions.